homeassistant/components/arlo/sensor.py

```diff
@@ -22,17 +28,43 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-# sensor_type [ description, unit, icon ]
-SENSOR_TYPES = {
-    "last_capture": ["Last", None, "run-fast"],
-    "total_cameras": ["Arlo Cameras", None, "video"],
-    "captured_today": ["Captured Today", None, "file-video"],
-    "battery_level": ["Battery Level", PERCENTAGE, "battery-50"],
-    "signal_strength": ["Signal Strength", None, "signal"],
-    "temperature": ["Temperature", TEMP_CELSIUS, "thermometer"],
-    "humidity": ["Humidity", PERCENTAGE, "water-percent"],
-    "air_quality": ["Air Quality", CONCENTRATION_PARTS_PER_MILLION, "biohazard"],
-}
+SENSOR_TYPES = (
+    SensorEntityDescription(key="last_capture", name="Last", icon="mdi:run-fast"),
+    SensorEntityDescription(key="total_cameras", name="Arlo Cameras", icon="mdi:video"),
+    SensorEntityDescription(
+        key="captured_today", name="Captured Today", icon="mdi:file-video"
+    ),
+    SensorEntityDescription(
+        key="battery_level",
+        name="Battery Level",
+        unit_of_measurement=PERCENTAGE,
+        icon="mdi:battery-50",
+        device_class=DEVICE_CLASS_BATTERY,
+    ),
+    SensorEntityDescription(
+        key="signal_strength", name="Signal Strength", icon="mdi:signal"
+    ),
+    SensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        unit_of_measurement=TEMP_CELSIUS,
+        icon="mdi:thermometer",
+        device_class=DEVICE_CLASS_TEMPERATURE,
```

Comment:
When a `device_class` is present, `unit_of_measurement` and `icon`  should not be.
```suggestion
        device_class=DEVICE_CLASS_TEMPERATURE,
```